### DISCUSSION TITLE
Are CalculationGroup and CalculationGroupTable really the same?

### DISCUSSION THREAD

**Discussion by bernatagulloesbrina** [TabularEditor/TabularEditor3#653] [🙋♀️TE3 Q&A] [answered]
Hi, I think on twitter you Daniel mensioned that CalculationGroup is an alias for CalculationGroupTable (or the other way around, can't remember). However, if that's the case why do I get such an error when coding in VisualStudio? 

![image](https://user-images.githubusercontent.com/40455069/189441270-0094eedb-c4f9-4a03-a433-b24e7f2e1c3f.png)

**Comment by ninmonkey** [comment]
![image](https://user-images.githubusercontent.com/3892031/189459608-5b150092-5373-4ed8-9b94-55f6b191b119.png)

They look different

- <https://docs.tabulareditor.com/api/TabularEditor.TOMWrapper.CalculationGroup.html>
- <https://docs.tabulareditor.com/api/TabularEditor.TOMWrapper.CalculationGroupTable.html>

**Reply by otykier** [reply, marked as answer]
Correct - they are two different objects. `CalculationGroupTable` represents the table. It's a special kind of table which holds a `CalculationGroup` object, which in turns holds the collection of `CalculationItem` objects.

@bernatagulloesbrina is probably thinking of [`Model.AddCalculationGroup(...)`](https://docs.tabulareditor.com/api/TabularEditor.TOMWrapper.Model.html#TabularEditor_TOMWrapper_Model_AddCalculationGroup_System_String_) and [`Model.AddCalculationGroupTable(...)`](https://docs.tabulareditor.com/api/TabularEditor.TOMWrapper.Model.html#TabularEditor_TOMWrapper_Model_AddCalculationGroupTable_System_String_) which do the same thing, and both return the `CalculationGroupTable`.

**Comment by bernatagulloesbrina** [comment]
I can't find the tweet -- but anyway, I'm a bit confused on when we should
use one or the other.
Normally I end up using CalculationGroupTable, but not sure on the theory
underneath

El sáb, 10 sept 2022 a las 1:48, Jake Bolton ***@***.***>)
escribió:

> [image: image]
> <https://user-images.githubusercontent.com/3892031/189459608-5b150092-5373-4ed8-9b94-55f6b191b119.png>
>
> They look different
>
>    -
>    https://docs.tabulareditor.com/api/TabularEditor.TOMWrapper.CalculationGroup.html
>    -
>    https://docs.tabulareditor.com/api/TabularEditor.TOMWrapper.CalculationGroupTable.html
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/TabularEditor/TabularEditor3/discussions/653#discussioncomment-3612492>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJUUXHKOFGASC3BJXDZUPLLV5PEF3ANCNFSM6AAAAAAQI75BNA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> .com>
>


-- 

Bernat Agulló

Partner & Data Enthusiast

***@***.***

+34 637 93 42 76
